src/plugins/intel_gpu/src/graph/input_layout.cpp

```diff
@@ -55,6 +55,13 @@ void input_layout_inst::set_data(memory::ptr mem) {
 
     _has_valid_input = true;
     _output_changed = true;
+    _shape_changed = mem->get_layout() != ol;
+
+    if (_shape_changed) {
+        node.invalidate_users();
```

Comment:
same above, this is not actually needed any more.

Reply:
duplicate comment. done.